platform-sdk/docs/proposals/consensus-layer/Consensus-Layer.md

```diff
@@ -0,0 +1,704 @@
+useful for testing or future maintenance tasks), and we also support circular dependencies between modules.
+
+### Foundational Concepts
+
+This design relies on several foundational concepts based on the hashgraph consensus algorithm. The term "hashgraph"
+refers to a data structure, while the "hashgraph consensus algorithm" refers to the particular consensus algorithm
+making use of a hashgraph for the purposes of consensus. The algorithm itself is only superficially described here,
+sufficient only to understand the overall system architecture.
+
+A hashgraph is a directed acyclic graph (or DAG), made up of _events_. Each event maintains references to "parent"
+events. When the hashgraph consensus algorithm runs, it collects events into _consensus rounds_. One or more rounds is
+grouped together by the Execution layer, executed, and used to form a block in the blockchain.
+
+Each event contains an ordered list of _transactions_.
+
+Nodes create events. Each event in the hashgraph has a _creator_. The creator is the node that created the event. Each
+event also has a _birth round_. This is the most recent round number known by the creator at the time it created the
+event. When a node creates an event, it fills that event with some, or all, of the transactions it knows about. Each
+creator creates a single event at a time, with some interval between event creations (say, every 50ms), and some maximum
+network-wide configuration for the number of events per second per creator. Each event will have as a "self-parent" the
+previous event created by that creator, assuming the creator remembers its previous event, and that event isn't ancient.
+Each event will additionally have one or more "other parent" events created by other creators, apart from the edge cases
+of network genesis and single node networks, where there may be zero other parents.
+
+Any given node has a system clock, and this clock provides the node with the current _wall clock time_. This is the
+current "real" time, as the node understands it. Since we cannot trust the clock of any particular node, we cannot trust
+the wall clock time of the creator to be accurate. The creator may lie. During the execution of the hashgraph consensus
+algorithm, each node will deterministically assign a _consensus time_ to each event (and thus by extension to each of
+the transactions within the event).
+
+Each node has a _roster_ listing all other nodes, their public cryptographic keys, their consensus weights (since the
+network is a proof-of-stake network, different nodes may have different "weights" when voting for consensus), etc. The
+cryptographic keys are used to verify that an event created by a creator was truly created by that creator. The roster
+can change over time, so it is vital that the correct roster be used for verifying each event. The correct roster to use
+for verifying an event is the roster that was active during that event's birth round.
+
+![Hashgraph](hashgraph.png)
```

Comment:
The arrows in this diagram are black and can't be seen on a black background. Probably we can change them to white.

Reply:
White arrows wouldn't be visible on a white background, though.

But fair point, so I added a white fill to the images, so they will be visible no matter what

Reply:
That will be even a more universal approach. Great, thank you!